video/out/vo_sixel.c

```diff
@@ -0,0 +1,483 @@
+/*
+ * video output driver for libsixel
+ * Based on the code for FFMPEG Sixel output driver by Hayaki Saito
+ *
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+#include <string.h>
+#include <time.h>
+#include <errno.h>
+#include <assert.h>
+
+#include <libavutil/rational.h>
+#include <libavutil/time.h>
+#include <libswscale/swscale.h>
+#include <sixel.h>
+#include <termios.h>
+
+#include "config.h"
+#include "options/m_config.h"
+#include "osdep/terminal.h"
+#include "sub/osd.h"
+#include "vo.h"
+#include "video/sws_utils.h"
+#include "video/mp_image.h"
+
+#include "config.h"
+#if !HAVE_GPL
```

Comment:
If the essence of this code is copied from elsewhere, then the license at the top should reflect that. You can't just take code which someone else wrote and claim it's owned by mpv.

You should check who the original copyright holder is, what the original license is, and ensure it's compatible with mpv (GPLv2 or later, or LGPL - check the mpv license docs).

Also, I don't _think_ we have such `HAVE_GPL` at mpv code, and generally it shouldn't be here.

Reply:
https://github.com/saitoha/FFmpeg-SIXEL/blob/sixel/libavdevice/sixel.c This is the reference implementation file, and it contains a standard FFMPEG LIcense (which also seems to be under same GNU General Lesser GPL (same as the vo_tct.c file).
So it should be okay right, if I remove the `HAVE_GPL` part and keep the same license?

Reply:
Well, the original file has `Copyright (c) 2014 Hayaki Saito` so at the very least it should stay at the file in mpv as well.

Preferably you should also add a link to the original file by the original author before the license.

Assuming the license is the same (please confirm) as in mpv, then you can keep the mpv license clause because the file and the code in it are now mpv-specific and part of mpv.

Reply:
Have updated the license file as per the suggestions.